src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -0,0 +1,224 @@
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\NamespacesAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\NamespaceUsesAnalyzer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author VeeWee <toonverwerft@gmail.com>
+ */
+final class FullyQualifiedStrictTypesFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Transforms imported FQCN parameters and return types to short version.',
+            [
+                new CodeSample(
+                    '<?php
+
+use Foo\Bar;
+
+class SomeClass
+{
+    public function doSomething(\Foo\Bar $foo)
+    {
+    }
+}
+'
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+
+use Foo\Bar;
+use Foo\Bar\Baz;
+
+class SomeClass
+{
+    public function doSomething(\Foo\Bar $foo): \Foo\Bar\Baz
+    {
+    }
+}
+',
+                    new VersionSpecification(70000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_FUNCTION);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $namespaces = array_map(function (NamespaceAnalysis $info) {
+            return $info->getFullName();
+        }, (new NamespacesAnalyzer())->getDeclarations($tokens));
+        $useMap = array_map(function (NamespaceUseAnalysis $info) {
+            return $info->getFullName();
+        }, (new NamespaceUsesAnalyzer())->getDeclarationsFromTokens($tokens));
+
+        if (!count($namespaces) && !count($useMap)) {
+            return;
+        }
+
+        $lastIndex = $tokens->count() - 1;
+        for ($index = $lastIndex; $index >= 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            // Return types are only available since PHP 7.0
+            if (PHP_VERSION_ID >= 70000) {
+                $this->fixFunctionReturnType($tokens, $index, $namespaces, $useMap);
+            }
+            $this->fixFunctionArguments($tokens, $index, $namespaces, $useMap);
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     * @param array  $namespaces
+     * @param array  $useMap
+     */
+    private function fixFunctionArguments(Tokens $tokens, $index, array $namespaces, array $useMap)
+    {
+        $arguments = (new FunctionsAnalyzer())->getFunctionArguments($tokens, $index);
+
+        foreach ($arguments as $argument) {
+            if (!$argument->hasType()) {
+                continue;
+            }
+
+            $shortType = $this->detectShortType($argument->getType(), $namespaces, $useMap);
+            if ($shortType === $argument->getType()) {
```

Comment:
Here comment would be nice, since short type is "the last word of imported namespace" and that might not be clear to other programmers, who will read this in following years.

E.g.

```php
// use Foo\Bar => "Bar"
```